packages/m365_defender/data_stream/event/elasticsearch/ingest_pipeline/pipeline_device.yml

```
@@ -2324,12 +2338,27 @@ processors:
       tag: set_dns_response_code
       ignore_empty_value: true
       if: ctx.event?.category != null && ctx.event.category.contains('network') && ctx.m365_defender?.event?.action?.type != null && ctx.m365_defender.event.action.type.toLowerCase().contains('dnsconnectioninspected')
-  - set:
-      field: dns.answers
-      copy_from: m365_defender.event.dns.answers
-      tag: set_dns_answers
-      ignore_empty_value: true
-      if: ctx.event?.category != null && ctx.event.category.contains('network') && ctx.m365_defender?.event?.action?.type != null && ctx.m365_defender.event.action.type.toLowerCase().contains('dnsconnectioninspected')
+  - script:
+      lang: painless
+      tag: set_dns_answers_and_ttls
+      if: "ctx.m365_defender?.event?.dns?.answers != null && ctx.m365_defender?.event?.dns?.ttls != null"
+      source: >-
+        def answers = ctx.m365_defender.event.dns.answers;
+        def ttls = ctx.m365_defender.event.dns.ttls;
+        if (answers.isEmpty() || ttls.isEmpty() || answers.length != ttls.length) {
```

I think in the case that `answers.length != ttls.length` we should probably write a message into `error.message` rather than silently dropping this.